platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -893,9 +935,8 @@ int _odp_ipsec_sa_replay_precheck(ipsec_sa_t *ipsec_sa, uint32_t seq,
 				  odp_ipsec_op_status_t *status)
 {
 	/* Try to be as quick as possible, we will discard packets later */
-	if (ipsec_sa->antireplay &&
-	    seq + IPSEC_ANTIREPLAY_WS <=
-	    (odp_atomic_load_u64(&ipsec_sa->hot.in.antireplay) & 0xffffffff)) {
+	if ((ipsec_sa->antireplay) && ((seq + ipsec_sa->ar.win_size) <=
+	    ipsec_sa->hot.in.wintop_seq)) {
```

Comment:
Reading ipsec_sa->hot.in.wintop_seq is a data race since window update running in another thread may be writing it at the same time. Using an atomic variable and atomic read would avoid the data race.

Reply:
ACK, will check atomic and also will check the usage of existing spinlock.